src/optimal_image/compress/image.rs

```diff
@@ -0,0 +1,53 @@
+
+use std::ffi::{CString, CStr};
+use std::error::Error;
+use std::sync::atomic::AtomicBool;
+use std::os::raw::{c_int, c_void, c_char};
+use std::ptr::null;
+use std::marker::PhantomData;
+use std::sync::atomic::Ordering::Relaxed;
+
+
+pub struct VipsImage<'a> {
+    pub c: *mut ffi::VipsImage,
+    marker: PhantomData<&'a()>,
+}
+
+impl<'a> Drop for VipsImage<'a> {
+    fn drop(&mut self) {
+        unsafe {
+            ffi::g_object_unref(self.c as *mut c_void);
+        }
+    }
+}
+
+impl<'a> VipsImage<'a> {
```

Comment:
@addityasingh to keep it easy we can try it out reading and saving to file every time, but of course that would be pretty slow and we need to be able to seamlessly use in memory buffers. 

The first challenge would be to make sure we can read and write from raw buffers both as `VipsImage` and our `ImageData` format (`ImgVec<RGBA>`)

The second challenge would probably be to please the borrow checker doing that :dancer:

Reply:
Agree. The borrow checker is a best in itself. But lets go ahead with the gradual approach and make it work. The in-memory buffers could be added as improvement once this works